pkg/bootkube/parse.go

```diff
@@ -0,0 +1,78 @@
+package bootkube
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/kubernetes-incubator/bootkube/pkg/asset"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	apiServerContainerName         = "kube-apiserver"
+	controllerManagerContainerName = "kube-controller-manager"
+)
+
+// detectServiceCIDR deserializes the '--service-cluster-ip-range' from the
+// kube-apiserver manifest
+func detectServiceCIDR(config Config) (string, error) {
+	b, err := ioutil.ReadFile(filepath.Join(config.AssetDir, asset.AssetPathAPIServer))
+	if err != nil {
+		return "", err
+	}
+	var apiServer v1beta1.DaemonSet
+	err = yaml.Unmarshal(b, &apiServer)
+	if err != nil {
+		return "", err
+	}
+
+	containers := map[string]v1.Container{}
+	for _, container := range apiServer.Spec.Template.Spec.Containers {
+		containers[container.Name] = container
+	}
+
+	if container, exists := containers[apiServerContainerName]; exists {
+		return findFlag("--service-cluster-ip-range", container.Command), nil
+	}
+	return "", fmt.Errorf("can't detect --service-cluster-ip-range in %s", asset.AssetPathAPIServer)
+}
+
+// detectPodCIDR deserializes the '--cluster-cidr' from the
+// kube-controller-manager manifest.
+func detectPodCIDR(config Config) (string, error) {
+	b, err := ioutil.ReadFile(filepath.Join(config.AssetDir, asset.AssetPathControllerManager))
+	if err != nil {
+		return "", err
+	}
+	var manager v1beta1.Deployment
+	err = yaml.Unmarshal(b, &manager)
+	if err != nil {
+		return "", err
+	}
+
+	containers := map[string]v1.Container{}
+	for _, container := range manager.Spec.Template.Spec.Containers {
+		containers[container.Name] = container
+	}
+
+	if container, exists := containers[controllerManagerContainerName]; exists {
+		return findFlag("--cluster-cidr", container.Command), nil
+	}
+	return "", fmt.Errorf("can't detect --cluster-cidr in %s", asset.AssetPathControllerManager)
+}
+
+func findFlag(flagName string, args []string) string {
+	for _, arg := range args {
+		if strings.HasPrefix(arg, flagName+"=") {
+			return strings.TrimPrefix(arg, flagName+"=")
+		}
+		if strings.HasPrefix(arg, flagName) {
+			return strings.TrimSpace(strings.TrimPrefix(arg, flagName))
+		}
+	}
+	return "unknown"
```

Comment:
This should be an error. Otherwise we're going to end up using something like `--cluster-cidr=unknown` and waiting for the bootstrap to fail - where we could have caught / surfaced here.